src/pages/library/Showcase.js

```diff
@@ -16,104 +18,333 @@ import * as Style from 'src/libs/style'
         width: 87,
         ...Utils.cond(
           [name.toLowerCase().includes('covid'), () => ({ backgroundImage: `url(${covidBg})` })],
           [variant === 'gatk', () => ({ backgroundColor: '#333', backgroundImage: `url(${gatkLogo})`, backgroundSize: undefined })],
           () => ({ backgroundImage: `url(${featuredBg})`, opacity: 0.75 })
         )
       }
-    })
+    }),
+    div({ style: { flex: 1, minWidth: 0, padding: '15px 20px', overflow: 'hidden' } }, [
+      div({ style: { display: 'flex' } }, [
+        div({ style: { flex: 1, color: colors.accent(), fontSize: 16, lineHeight: '20px', height: 40, marginBottom: 7 } }, [name]),
+        created && div([Utils.makeStandardDate(created)])
+      ]),
+      div({ style: { lineHeight: '20px', height: 100, whiteSpace: 'pre-wrap', overflow: 'hidden' } }, [description])
+      // h(MarkdownViewer, [description]) // TODO: should we render this as markdown?
+    ])
   ])
 }
 
+// collapsible section for sidebar categories
+const SidebarCollapser = ({ title, isOpened, onClick, children }) => {
+  return div({
+    role: 'group'
+  }, [
+    h(NavItem, {
+      onClick,
+      style: { ...styles.sidebarRow, ...styles.nav.navSection }
+    }, [
+      title,
+      div({ style: { flexGrow: 1 } }),
+      icon(isOpened ? 'angle-up' : 'angle-down', { size: 18, style: { flex: 'none' } })
+    ]),
+    div({
+      style: { flex: 'none' }
+    }, [h(RCollapse, { isOpened }, [children])])
+  ])
+}
+
+const groupByFeaturedTags = workspaces => _.flow([
+  _.map(tag => [tag, _.filter(w => _.includes(tag, w.tags.items), workspaces)]),
+  _.fromPairs
+])(uniqueSidebarTags)
+
+const Pill = ({ count, highlight }) => {
+  return div({
+    style: _.merge(styles.pill, highlight ? styles.pillHighlight : {})
+  }, [count])
+}
+
+const Sidebar = ({ onSectionFilter, onTagFilter, sections, selectedSections, selectedTags, workspacesByTag }) => {
+  const [collapsedSections, setCollapsedSections] = useState([])
+
+  const unionSectionWorkspaces = section => {
+    return _.uniq(_.flatMap(tag => workspacesByTag[tag], section.tags))
+  }
+
+  return div({ style: { display: 'flex', flexDirection: 'column' } }, [
+    _.map(section => {
+      return section.keepCollapsed ?
+        h(Clickable, {
+          key: section.name,
+          onClick: () => onSectionFilter(section),
+          style: { ...styles.sidebarRow, ...styles.nav.navSection }
+        }, [
+          div({ style: { flex: 1 } }, [section.name]),
+          h(Pill, {
+            count: _.size(unionSectionWorkspaces(section)),
+            highlight: _.includes(section, selectedSections)
+          })
+        ]) :
+        h(SidebarCollapser,
+          {
+            key: section.name,
+            title: section.name,
+            onClick: () => setCollapsedSections(_.xor([section], collapsedSections)),
+            isOpened: !_.includes(section, collapsedSections)
+          },
+          [
+            ..._.map(label => {
```

Comment:
instead of spread merging this array, could you just do the map directly and use a key on the clickable?

Reply:
Interesting. Recognizing that `_.map` would return an array, I removed the surrounding `[]` and... react didn't complain about not specifying `key`s. Sure enough, the same is true for the surrounding map over `sections`.

Should I remove the `[]`s and add/keep keys or remove all of it? From my reading, it looks like keeping them in an array and providing keys might be more performant.

Reply:
yup, your target is:
```js
div([
  _.map(x => div({ key: x.uniqueId }), list)
])
```